### PR TITLE
added javadocs + more methods to DynamoGearProvider

### PR DESCRIPTION
**Pull request by nathanbaney**
added junit tests for issue6 and 7, basic adaptor/internal request model usage

### CODE REVIEW

File: caampr-backend/src/main/java/com/jelistan/caampr/lambda/adaptor/GearListRequestAdaptor.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+/**
+ * Converts APIGatewayProxyRequestEvent (external) to GearListRequest (internal)
+ */
+
+public class GearListRequestAdaptor{
+    public GearListRequest convert(APIGatewayProxyRequestEvent requestEvent){
+        String[] tokens = requestEvent.getPath().split("/");
+        String profileId = tokens[2];
+        GearListRequest request = GearListRequest.builder()
+                .callerId("6969") //temporary hardcode until requester info is implemented
```
Comment (jebaney) [inline caampr-backend/src/main/java/com/jelistan/caampr/lambda/adaptor/GearListRequestAdaptor.java]: just an FYI (meaning you don't have to do it here): normally, if you are leaving a stub or placeholder, add a TODO comment and create an associated issue/ticket to track that the TODO is resolved. e.g.
```
// TODO hardcoding the callerId until it's passed in for real. tracking in issue [link to issue]
```

File: caampr-backend/src/main/java/com/jelistan/caampr/lambda/handler/GetGearListHandler.java [hunk 2]
```diff
@@ -27,8 +29,10 @@ public GetGearListHandler(final GearProvider gearProvider) {
 
     @Override
     public APIGatewayProxyResponseEvent handleRequest(final APIGatewayProxyRequestEvent event, Context context)  {
+        GearListRequestAdaptor adaptor = new GearListRequestAdaptor();
```
Comment (jebaney) [inline caampr-backend/src/main/java/com/jelistan/caampr/lambda/handler/GetGearListHandler.java]: The adapter should be created in the dagger module (probably `ProviderModule`) and injected via the constructor

File: caampr-backend/src/main/java/com/jelistan/caampr/lambda/model/Profile.java [hunk 3]
```diff
@@ -1,14 +1,20 @@
 package com.jelistan.caampr.lambda.model;
 
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBAttribute;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBHashKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTable;
 import lombok.Builder;
 import lombok.Data;
 
+@DynamoDBTable(tableName = "Gear-v1")
```
Comment (jebaney) [inline caampr-backend/src/main/java/com/jelistan/caampr/lambda/model/Profile.java]: The profile model should live in its own table and not in the gear table.  So something like `Profile-v1`

File: caampr-backend/src/main/java/com/jelistan/caampr/lambda/model/Profile.java [hunk 4]
```diff
@@ -1,14 +1,20 @@
...
 @Builder
 @Data
 public class Profile {
+    @DynamoDBHashKey(attributeName = "id")
```
Comment (jebaney) [inline caampr-backend/src/main/java/com/jelistan/caampr/lambda/model/Profile.java:12]: The profileId is the id for this table (key), so update `id` to be a string and get rid of the `profileName` field.  Make sense?  Normally, I would have a data model already documented for this, so you're kind of flying blind, so my fault

File: caampr-backend/src/main/java/com/jelistan/caampr/lambda/provider/GearListProvider.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClientBuilder;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBQueryExpression;
+import com.jelistan.caampr.lambda.model.Gear;
+import com.jelistan.caampr.lambda.model.GearListRequest;
+
+import java.util.List;
+
+public class GearListProvider implements GearProvider{
```
Comment (jebaney) [inline caampr-backend/src/main/java/com/jelistan/caampr/lambda/provider/GearListProvider.java]: What is this class for?  We already have a dynamodb implementation.

File: caampr-backend/src/test/java/com/jelistan/caampr/lambda/adaptor/GearListRequestAdaptorTest.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+import org.mockito.MockitoAnnotations;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.when;
+
+public class GearListRequestAdaptorTest {
+    @Mock
+    private APIGatewayProxyRequestEvent requestEvent;
+
+    private GearListRequestAdaptor adaptor;
```
Comment (jebaney) [inline caampr-backend/src/test/java/com/jelistan/caampr/lambda/adaptor/GearListRequestAdaptorTest.java]: For this project, the class under test should be named `unit`.  This helps to identify things

File: caampr-backend/src/test/java/com/jelistan/caampr/lambda/provider/DynamoGearProviderTest.java [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+    @Mock
+    private PaginatedQueryList<Gear> queryList;
+
+    private DynamoGearProvider unit;
+
+    @Before
+    public void setUp() throws Exception {
+        MockitoAnnotations.openMocks(this);
+        queryList.add(GearProvider.FAKE_GEAR);
```
Comment (jebaney) [inline caampr-backend/src/test/java/com/jelistan/caampr/lambda/provider/DynamoGearProviderTest.java]: `queryList` is a mock, so doing a `queryList.add` here doesn't serve any purpose.

**Review by jebaney — CHANGES_REQUESTED** [review]

File: caampr-backend/src/main/java/com/jelistan/caampr/lambda/adaptor/GearListRequestAdaptor.java (hunk 1 above)
Comment (nathanbaney) [reply caampr-backend/src/main/java/com/jelistan/caampr/lambda/adaptor/GearListRequestAdaptor.java]: will do, make issue 18 for this

File: caampr-backend/src/main/java/com/jelistan/caampr/lambda/handler/GetGearListHandler.java (hunk 2 above)
Comment (nathanbaney) [reply caampr-backend/src/main/java/com/jelistan/caampr/lambda/handler/GetGearListHandler.java]: gotcha will do

File: caampr-backend/src/main/java/com/jelistan/caampr/lambda/model/Profile.java (hunk 3 above)
Comment (nathanbaney) [reply caampr-backend/src/main/java/com/jelistan/caampr/lambda/model/Profile.java]: ah gotcha, thought they shared the table and the profile just served as a secondary index or something, will do

File: caampr-backend/src/main/java/com/jelistan/caampr/lambda/model/Profile.java (hunk 4 above)
Comment (nathanbaney) [reply caampr-backend/src/main/java/com/jelistan/caampr/lambda/model/Profile.java:12]: gotcha, previous basically had 2 id fields for no reason, changed

File: caampr-backend/src/main/java/com/jelistan/caampr/lambda/provider/GearListProvider.java (hunk 5 above)
Comment (nathanbaney) [reply caampr-backend/src/main/java/com/jelistan/caampr/lambda/provider/GearListProvider.java]: gonna be real, i have no idea lol, maybe i made it before when i was trying to have all providers be inheritance based or something? i cant remember. ill delete and see if it breaks

File: caampr-backend/src/test/java/com/jelistan/caampr/lambda/adaptor/GearListRequestAdaptorTest.java (hunk 6 above)
Comment (nathanbaney) [reply caampr-backend/src/test/java/com/jelistan/caampr/lambda/adaptor/GearListRequestAdaptorTest.java]: changed

File: caampr-backend/src/test/java/com/jelistan/caampr/lambda/provider/DynamoGearProviderTest.java (hunk 7 above)
Comment (nathanbaney) [reply caampr-backend/src/test/java/com/jelistan/caampr/lambda/provider/DynamoGearProviderTest.java]: layover from when i understood mockito less, removed

**Pull request merged**